lua/basic_command_register.lua

```diff
@@ -0,0 +1,27 @@
+local win = vis.win
+vis:command_register("foo", function(argv, force, win, cursor, range)
+	append(win, '$', 'args: ' .. #argv)
+	for i,arg in ipairs(argv) do
+		append(win, '$', '\t' .. i .. ' ' .. arg)
+	end
+	append(win, '$', 'bang: ' .. (force and "yes" or "no"))
+	append(win, '$', 'name: ' .. win.file.name)
+	append(win, '$', 'pos: ' .. cursor.pos)
+	append(win, '$', 'range: ' .. (range ~= nil and
+				('['..range.start..', '..range.finish..']')
+				or "invalid range"))
+	append(win, '$', '')
+	return true;
+end)
+
+vis:command('foo')
+vis:command('foo!')
+-- TODO Providing a range does not work
```

Comment:
Sorry, you are right, the range works (as documented, it returns ranges as bytes), but it looks like the cursor position from `cursor.pos` is not always correct. If I do a `vis.win.cursor:to(1,1)` before one of the tests, the position reported by the test does not change. Am I using it wrong?